tensorflow_graphics/projects/pointnet/helpers.py

```diff
@@ -154,3 +161,106 @@ def handle_keyboard_interrupt(flags):
   else:
     print("Delete these summaries with: ")
     termcolor.cprint("  rm -rf {}".format(flags.logdir), "red")
+
+
+def preprocess(points,
+               labels,
+               num_points,
+               augment_jitter=False,
+               augment_rotate=False):
+  """
+  Preprocess point cloud data.
+
+  Example usage:
+  ```python
+  ds_train, ds_test = ModelNet40.load(
+      as_supervised=True, split=("train", "test")
+
+  num_points = 1024
+  ds_train = ds_train.map(
+      lambda points, labels: preprocess(points, labels, num_points, True, True))
+  ```
+
+  Args:
+    points: [N, 3] input pointcloud coordinates
+    labels: presumably scalar int labels, but is left untouched.
+    num_points: number of points to take from `points`
+    augment_jitter: bool indicating whether to jitter points randomly
+    augment_rotate: bool indicating whether to rotate points about z-axis
+      randomly
+
+  Returns:
+    out_points: [num_points, 3] potentially augmented point cloud
+    labels: same as input.
+  """
+  points = points[:num_points]
+  if augment_rotate:
+    points = augment.rotate(points)
+  if augment_jitter:
+    points = augment.jitter(points)
+  return points, labels
+
+
+def get_modelnet40_datasets(num_points=2048,
```

Comment:
move elsewhere? (file to remain project agnostic)

Reply:
In fact, there's nothing wrong with repeating the logic twice.
The "custom training loop" (my version) vs. keras (your version) are supposed to be self-contained examples. 

Being examples, sharing logic outside the file makes them less clean :)